app/models/ability.rb

```diff
@@ -0,0 +1,18 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize user
+    can :read, [Product, ProductDetail, Comment, Rate]
+    can :create, User
+    return unless user&.active_for_authentication?
+
+    can :manage, Cart, user_id: user.id
+    can %i(create read), Order, user_id: user.id
+    can :manage, Order, user_id: user.id, status: :pending
+    can :manage, User, id: user.id
+    can :manage, [Comment, Rate], user_id: user.id
+    return unless user.admin?
+
+    can :manage, :all
```

Comment:
Chị nghĩ nên đẩy chỗ check admin lên trc nhé 
`return can :manage, :all if user.admin?`
các line 9 - 13 sẽ đưa xún sau

Reply:
Công nhận đưa lên hay hơn, nhưng nó bị như này `W: Lint/ReturnInVoidContext: Do not return a value in initialize.`, để em xem có cách nào ko

Reply:
Để em dùng: 
```
if user.admin?
      can :manage, :all
      return
end
```